tests/unittests/core/io/test_space_builder.py

```diff
@@ -127,6 +127,17 @@ def test_build_fails_because_of_ValueError_on_init(self, dimbuilder):
         assert "Parameter" in str(exc.value)
         assert "size" in str(exc.value.__cause__)
 
+    def test_build_fails_because_of_invalid_arguments_for_custom_prior(
+        self, dimbuilder
+    ):
+        """Build fails because ValueError happens on init of builtin prior."""
+        with pytest.raises(ValueError) as exc:
```

Comment:
You could use the `match` argument of `raises` instead of checking if the value is in the exception message, I think.
(That might be only true for `pytest.warns` though).

Reply:
Indeed. This test is looking at two different parts of `exc.value` however.